### PR TITLE
None json throw exception instead of returning a left class in Url pa…

### PR DESCRIPTION
**Pull request by chitang**
…rsing.

### CODE REVIEW

File: soql-types/src/main/scala/com/socrata/soql/types/SoQLType.scala [hunk 1]
```diff
@@ -390,25 +388,45 @@ case class SoQLUrl(@JsonKey("url") url: Option[String],
...
+    parseAsJson(value).orElse(parseAsUri(value)).orElse(parseAsHtmlTag(value))
+  }
+
+  private def parseAsJson(value: String): Option[SoQLUrl] = {
     try {
-      if (JsonUtil.parseJson[SoQLUrl](s).isRight) { true }
-      else {
-        new URI(s)
-        true
+      JsonUtil.parseJson[SoQLUrl](value) match {
```
Comment (rjmac) [inline soql-types/src/main/scala/com/socrata/soql/types/SoQLType.scala]: `.right.toOption` is a fair bit shorter...

File: soql-types/src/main/scala/com/socrata/soql/types/SoQLType.scala [hunk 2]
```diff
@@ -390,25 +388,45 @@ case class SoQLUrl(@JsonKey("url") url: Option[String],
...
+    } catch {
+      case e: URISyntaxException => None
+    }
+  }
+
+  private def parseAsHtmlTag(value: String): Option[SoQLUrl] = {
+    try {
+      xml.XML.loadString(value) match {
+        case a @ <a>{_*}</a> =>
+          Some(SoQLUrl(a.attribute("href").map(_.text).orElse(None),
```
Comment (rjmac) [inline soql-types/src/main/scala/com/socrata/soql/types/SoQLType.scala]: `orElse(None)` doesn't do anything, unless this is isn't `Option#orElse`.

File: soql-types/src/main/scala/com/socrata/soql/types/SoQLType.scala (hunk 1 above)
Comment (chitang) [inline soql-types/src/main/scala/com/socrata/soql/types/SoQLType.scala]: fixed

**Comment by rjmac** [comment]
lgtm

**Pull request merged**